Mathlib/Data/Matrix/Basis.lean

```diff
@@ -68,16 +68,14 @@ theorem mulVec_stdBasisMatrix [NonUnitalNonAssocSemiring α] [Fintype m]
   · simp
   simp [h, h.symm]
 
+-- In the following proof, we abuse the definition of `Matrix` by applying matrices.
+-- We therefore need to parenthesize `Fintype.sum_apply` to convince `simp` to apply it even though
+-- the type is `Matrix m n α`, not `m → _`.
 theorem matrix_eq_sum_stdBasisMatrix [AddCommMonoid α] [Fintype m] [Fintype n] (x : Matrix m n α) :
     x = ∑ i : m, ∑ j : n, stdBasisMatrix i j (x i j) := by
-  ext i j; symm
-  iterate 2 rw [Finset.sum_apply]
-  convert (Fintype.sum_eq_single i ?_).trans ?_; swap
-  · -- Porting note(#12717): `simp` seems unwilling to apply `Fintype.sum_apply`
-    simp (config := { unfoldPartialApp := true }) [stdBasisMatrix, (Fintype.sum_apply)]
-  · intro j' hj'
-    -- Porting note(#12717): `simp` seems unwilling to apply `Fintype.sum_apply`
-    simp (config := { unfoldPartialApp := true }) [stdBasisMatrix, (Fintype.sum_apply), hj']
+  ext i j
+  rw [← Fintype.sum_prod_type']
+  simp [stdBasisMatrix, (Fintype.sum_apply), Matrix.of_apply, ← Prod.mk.inj_iff]
```

Comment:
I think the comment should stay next to the tactic in question, not next to the proof.